From 2b36a74ce1c0cb11103377ba8a70a10850927f28 Mon Sep 17 00:00:00 2001
From: Alan Tull <r80115@freescale.com>
Date: Thu, 5 Feb 2009 13:55:49 -0600
Subject: [PATCH] ENGR00107223-2 SOC: fail soc_match_components if mach_probe fails

If the machine driver's mach_probe fails, that can indicate
that the machine driver's codec is not present on the board.

Signed-off-by: Alan Tull <r80115@freescale.com>
---
 sound/soc/soc-core.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 962e262..0ca0089 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -837,7 +837,7 @@ static int soc_match_components(void)
 
 	/* are all pcm_links now created ? */
 	if (probe && machine && machine->pcm_links == machine->pcm_links_total)
-		machine->ops->mach_probe(machine);
+		ret = machine->ops->mach_probe(machine);
 	mutex_unlock(&list_mutex);
 	return ret;
 }
-- 
1.5.4.4

